tools/render-test/d3d-util.cpp

```diff
@@ -0,0 +1,280 @@
+﻿// d3d-util.cpp
+#include "d3d-util.h"
+
+#include <d3dcompiler.h>
+
+// We will use the C standard library just for printing error messages.
+#include <stdio.h>
+
+namespace renderer_test {
+using namespace Slang;
+
+/* static */D3D_PRIMITIVE_TOPOLOGY D3DUtil::getPrimitiveTopology(PrimitiveTopology topology)
+{
+    switch (topology)
+    {
+        case PrimitiveTopology::TriangleList:
+        {
+            return D3D11_PRIMITIVE_TOPOLOGY_TRIANGLELIST;
+        }
+        default: break;
+    }
+    return D3D11_PRIMITIVE_TOPOLOGY_UNDEFINED;
+}
+
+/* static */DXGI_FORMAT D3DUtil::getMapFormat(Format format)
+{
+    switch (format)
+    {
+        case Format::RGB_Float32:
+            return DXGI_FORMAT_R32G32B32_FLOAT;
+        case Format::RG_Float32:
+            return DXGI_FORMAT_R32G32_FLOAT;
+        default:
+            return DXGI_FORMAT_UNKNOWN;
+    }
+}
+
+/* static */DXGI_FORMAT D3DUtil::calcResourceFormat(UsageType usage, Int usageFlags, DXGI_FORMAT format)
+{
+    SLANG_UNUSED(usage);
+    if (usageFlags)
+    {
+        switch (format)
+        {
+            case DXGI_FORMAT_R32_FLOAT:
+            case DXGI_FORMAT_D32_FLOAT:            return DXGI_FORMAT_R32_TYPELESS;
+            case DXGI_FORMAT_D24_UNORM_S8_UINT:    return DXGI_FORMAT_R24G8_TYPELESS;
+            default: break;
+        }
+        return format;
+    }
+    return format;
+}
+
+/* static */DXGI_FORMAT D3DUtil::calcFormat(UsageType usage, DXGI_FORMAT format)
+{
+    switch (usage)
+    {
+        case USAGE_COUNT_OF:
+        case USAGE_UNKNOWN:
+        {
+            return DXGI_FORMAT_UNKNOWN;
+        }
+        case USAGE_DEPTH_STENCIL:
+        {
+            switch (format)
+            {
+                case DXGI_FORMAT_D32_FLOAT:
+                case DXGI_FORMAT_R32_TYPELESS:            return DXGI_FORMAT_D32_FLOAT;
```

Comment:
Sure. There's a couple of ways that could perhaps help. Could always brace an action that is for multiple cases. Could also use /* fallthru */ comment to make clear the intention.
Well I tried it - and so used braces with all. I only put fallthru if small amount of cases - because the big block of cases, makes pretty clear whats going on, whereas just a couple can get lost in the noise.